internal/policies/mount/internal_test.go

```diff
@@ -0,0 +1,135 @@
+package mount
+
+import (
+	"flag"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/adsys/internal/testutils"
+	"gopkg.in/yaml.v3"
+)
+
+var Update bool
+
+func TestParseEntryValues(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		entry string
+	}{
+		// Single entry cases.
+		"parse values from entry with one value":        {entry: "entry with one value"},
+		"parse values from entry with multiple values":  {entry: "entry with multiple values"},
+		"parse values from entry with repeatead values": {entry: "entry with repeatead values"},
+
+		// Badly formatted entries.
+		"parse values trimming whitespaces":           {entry: "entry with spaces"},
+		"parse values trimming sequential linebreaks": {entry: "entry with multiple linebreaks"},
+
+		// Special cases.
+		"parse values from entry with anonymous tags": {entry: "entry with anonymous tags"},
+		"returns empty slice if the entry is empty":   {entry: "entry with no value"},
+	}
+
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			got := parseEntryValues(EntriesForTests[tc.entry])
+
+			gotPath := t.TempDir()
+			marshaledGot, err := yaml.Marshal(got)
+			require.NoError(t, err, "Setup: Failed to marshal the result")
+
+			err = os.WriteFile(filepath.Join(gotPath, "parsed_values"), marshaledGot, 0600)
+			require.NoError(t, err, "Setup: Failed to write the result")
+
+			goldenPath := filepath.Join("testdata", t.Name(), "golden")
+			testutils.CompareTreesWithFiltering(t, gotPath, goldenPath, Update)
+		})
+	}
+}
+
+func TestWriteFileWithUIDGID(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		uid     string
+		gid     string
+		content string
+
+		readOnlyDir       bool
+		pathAlreadyExists bool
+
+		wantErr bool
+	}{
+		"write file with current user ownership": {},
+
+		"error when invalid uid":                               {uid: "-150", wantErr: true},
+		"error when invalid gid":                               {gid: "-150", wantErr: true},
+		"fails when writing on a dir with invalid permissions": {readOnlyDir: true, wantErr: true},
```

Comment:
I would prefix again with `error` here, so that we can run all error cases with `go test -run TestWriteFileWithUIDGID/error`